test/unit/int_algolia/scripts/algolia/model/algoliaLocalizedProduct.test.js

```diff
@@ -504,3 +504,59 @@ describe('algoliaLocalizedProduct overriding custom attributes', function () {
         expect(new AlgoliaLocalizedProductModel({ product: product, locale: 'fr', attributeList: ['algoliaTest', 'custom.displaySize'], baseModel: baseModel})).toEqual(expectedProductModel);
     });
 });
+
+describe('IndexOutofStock logic tests', function () {
+    var ProductMockObj;
+
+    beforeEach(() => {
+        jest.resetModules();
+        ProductMockObj = require('../../../../../mocks/dw/catalog/Variant');
+    });
+
+    test('should skip out-of-stock variants if IndexOutofStock = false', function () {
+        jest.doMock('*/cartridge/scripts/algolia/lib/algoliaData', () => {
+            return {
+                getSetOfArray: function () {
+                    return []; 
+                },
+                getPreference: function (id) {
+                    if (id === 'InStockThreshold') return 1;
+                    if (id === 'IndexOutofStock') return false;
+                    return null;
+                }
+            };
+        }, { virtual: true });
+
+
+        const product = new ProductMockObj({
+            inventory: { atsValue: 0 }
+        });
+
+        const localizedProduct = new AlgoliaLocalizedProduct({ product, attributeList: ['objectID'] });
+
+        expect(localizedProduct.objectID).toBeNull();
+    });
+
+    test('should index out-of-stock variants if IndexOutofStock = true', function () {
+        jest.doMock('*/cartridge/scripts/algolia/lib/algoliaData', () => {
+            return {
+                getSetOfArray: function () {
+                    return []; 
+                },
+                getPreference: function (id) {
+                    if (id === 'InStockThreshold') return 1;
+                    if (id === 'IndexOutofStock') return true; 
+                    return null;
+                }
+            };
+        }, { virtual: true });
+
+        const product = new ProductMockObj({
+            inventory: { atsValue: 0 }
+        });
+
+        const localizedProduct = new AlgoliaLocalizedProduct({ product, attributeList: ['objectID'] });
+
+        expect(localizedProduct).not.toBeNull();
```

Comment:
Same, there is no relation with indexing out-of-stocks products at all.
You are instantiating a `new AlgoliaLocalizedProduct`, so it's normal for it to not be null